workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java

```diff
@@ -82,6 +83,11 @@ public WorkReport executeAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, W
 
 		/* get workflow definition entity */
 		WorkFlowDefinition workFlowDefinition = this.workFlowDefinitionRepository.findFirstByName(workflowName);
+		if (workFlowDefinition == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new WorkflowDefinitionNotFoundException("Cannot find workflow '" + workflowName + "'"));
```

Comment:
recently I made some refactor around the exceptions and we can use it here instead:
```
new ResourceNotFoundException(ResourceType.WORKFLOW_DEFINITION, IDType.NAME, workflowName)
```

Reply:
fixed